datastream-server/src/test/java/com/linkedin/datastream/server/assignment/TestBroadcastStrategy.java

```diff
@@ -279,4 +280,21 @@ public void testRebalanceTasksWhenNewInstanceIsAdded() {
 
     Assert.assertEquals(newAssignment.get(instance4).size(), expectedNumTasksPerInstance);
   }
+
+  @Test
+  public void testDontCreateNewTasksWhenInstanceChanged() {
+    String[] instances = new String[]{"instance1", "instance2", "instance3"};
+    List<DatastreamGroup> datastreams = generateDatastreams("ds", 5);
+    BroadcastStrategy strategy = new BroadcastStrategy(Optional.empty());
+    Map<String, Set<DatastreamTask>> assignment =
+        strategy.assign(datastreams, Arrays.asList(instances), new HashMap<>());
+    instances = new String[]{"instance1", "instance2", "instance4"};
+
+    Map<String, Set<DatastreamTask>> newAssignment = strategy.assign(datastreams, Arrays.asList(instances), assignment);
+    Set<DatastreamTask> oldAssignmentTasks = assignment.values().stream().flatMap(Set::stream).collect(Collectors.toSet());
+    Set<DatastreamTask> newAssignmentTasks = newAssignment.values().stream().flatMap(Set::stream).collect(Collectors.toSet());
+    LOG.info("{}", oldAssignmentTasks);
+    LOG.info("{}", newAssignmentTasks);
+    Assert.assertTrue(oldAssignmentTasks.containsAll(newAssignmentTasks));
```

Comment:
1. Should we make this assert stricter: assert they're actually the exact same set of tasks?

```java
Assert.assertEquals(oldAssignmentTasks, newAssignmentTasks);
```

2. Should we also assert that no task was assigned to more than one instance?

Reply:
Done.